packages/extension-code-block/src/code-block.ts

```diff
@@ -260,8 +260,15 @@ export const CodeBlock = Node.create<CodeBlockOptions>({
 
             const { tr } = view.state
 
-            // create an empty code block
-            tr.replaceSelectionWith(this.type.create({ language }))
+            // create an empty code block´
+            // if the cursor is at the absolute end of the document, insert the code block before the cursor instead
+            // of replacing the selection as the replaceSelectionWith function will cause the insertion to
+            // happen at the previous node
+            if (view.state.selection.from === view.state.doc.nodeSize - (1 + (view.state.selection.$to.depth * 2))) {
```

Comment:
I'm not going to pretend that I completely understand this. What is tripping me up is the `* 2`.

But otherwise this looks right

Reply:
The `2` is the offset of a prosemirror node to the next one. If you for example have a nested node - the wrapping node will always add an offset of 1 at the beginning and the end which you have to account for if you want to find the correct position after said node.